### ISSUE TITLE
Secrets with namespace scope overwritten by secrets with project scope without warning

### ISSUE THREAD

**Issue by mattmattox** [rancher/rancher#23050]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
1. Click Add Secret.
2. Enter a Name (user-secret) for the secret.
3. Select a Scope for the secret. Make the secret available for the specific namespace
4. From Secret Values, click Add Secret Value to add a key value pair. Add as many values as you need.
Repeat the same steps with same name and just change the scope for the entire project.

**Result:**
It does not give any error rather overwrites existing secret and UI shows two secrets until you refresh the UI.

**Other details that may be helpful:**
If reverse the order, it will give you an error.
1. Click Add Secret.
2. Enter a Name (user-secret) for the secret.
3. Select a Scope for the secret. Make the secret available for the entire project
4. From Secret Values, click Add Secret Value to add a key value pair. Add as many values as you need.
Repeat the same steps with same name and just change the scope as specific namespace.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):v2.2.7
- Installation option (single install/HA): HA

**Comment by westlywright** [comment]
I believe this is a API issue. I was able to duplicate and OP is right, if you do this in reverse order the API sends an error. The API should error if we try to create a namespaced secret with a name matching a non-namespaced secret since it's reverse is not a UI handled error. If this is allowed, the API should not overwrite the non-namespaced secret at the very least.

**Comment by izaac** [comment]
Reproduced in Rancher `v2.6.4` using the Ember UI.

**Comment by SheilaghM** [comment]
The behavior the customer is seeing is by design because they are promoting the namespace secret to a project secret and as a project owner you are allowed to override secrets within the namespaces of your project. Whereas doing the opposite (create project secret, then namespace secret with the same name) is not allowed and will throw an error because project secrets have precedence and shouldn’t be overridden by namespace secrets.

In short, Rancher's logic is that project secrets have precedence over namespace secrets so if a namespace and project have the same secret name, the project secret will always overwrite the namespace secret by design.
